### PR TITLE
Do some followup porting to react-18 react-dom client API

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

File: pkg/lib/cockpit-components-context-menu.jsx [hunk 1]
```diff
@@ -33,83 +33,83 @@ const _ = cockpit.gettext;
...
+        };
+
+        const parent = document.getElementById(parentId);
+        parent.addEventListener('contextmenu', _handleContextMenu);
+        document.addEventListener('click', _handleClick);
+
+        return () => {
+            parent.removeEventListener('contextmenu', _handleContextMenu);
+            document.removeEventListener('click', _handleClick);
+        };
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-context-menu.jsx:65]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18017-20221212-134212-cbdc492c-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-context-menu.jsx [hunk 2]
```diff
@@ -33,83 +33,83 @@
...
         const bottom = !top;
 
         if (right) {
-            this.root.style.left = `${clickX + 5}px`;
+            root.current.style.left = `${clickX + 5}px`;
         }
 
         if (left) {
-            this.root.style.left = `${clickX - rootW - 5}px`;
+            root.current.style.left = `${clickX - rootW - 5}px`;
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-context-menu.jsx:89]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18017-20221212-134212-cbdc492c-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-context-menu.jsx [hunk 3]
```diff
@@ -33,83 +33,83 @@
...
         }
 
         if (top) {
-            this.root.style.top = `${clickY + 5}px`;
+            root.current.style.top = `${clickY + 5}px`;
         }
 
         if (bottom) {
-            this.root.style.top = `${clickY - rootH - 5}px`;
+            root.current.style.top = `${clickY - rootH - 5}px`;
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-context-menu.jsx:97]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18017-20221212-134212-cbdc492c-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-context-menu.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-context-menu.jsx:65]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18017-20221212-220701-ab71eae1-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-context-menu.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-context-menu.jsx:89]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18017-20221212-220701-ab71eae1-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-context-menu.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-context-menu.jsx:97]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18017-20221212-220701-ab71eae1-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by marusak — APPROVED** [review]
Thank you!

File: pkg/lib/cockpit-components-context-menu.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-context-menu.jsx:65]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18017-20221213-083151-ab71eae1-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-context-menu.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-context-menu.jsx:89]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18017-20221213-083151-ab71eae1-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-context-menu.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-context-menu.jsx:97]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18017-20221213-083151-ab71eae1-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-context-menu.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-context-menu.jsx:65]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18017-20221213-100721-ab71eae1-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-context-menu.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-context-menu.jsx:89]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18017-20221213-100721-ab71eae1-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-context-menu.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-context-menu.jsx:97]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18017-20221213-100721-ab71eae1-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**